backend/cfg/llvmize.ml

```diff
@@ -995,29 +1006,42 @@ module F = struct
      [data_item] *)
   let typ_of_data_item (d : Cmm.data_item) =
     match d with
-    | Cdefine_symbol _ -> assert false (* cannot happen *)
-    | Cint _ -> Llvm_typ.Int { width_in_bits = 64 }
+    | Cdefine_symbol _ | Calign _ | Csymbol_offset _ ->
+      (* [Calign] and [Csymbol_offset] are never produced *)
+      Misc.fatal_error "Llvmize: unexpected data item"
+    | Cint _ -> Llvm_typ.i64
+    | Cint8 _ -> Llvm_typ.i8
+    | Cint16 _ -> Llvm_typ.i16
+    | Cint32 _ -> Llvm_typ.i32
     | Csymbol_address _ -> Llvm_typ.Ptr
-    | Cint8 _ | Cint16 _ | Cint32 _ | Csingle _ | Cdouble _ | Cvec128 _
-    | Cvec256 _ | Cvec512 _ | Csymbol_offset _ | Cstring _ | Cskip _ | Calign _
-      ->
-      Misc.fatal_error "Llvmize.typ_of_data_item: not implemented"
+    | Cstring s ->
+      Llvm_typ.Array { size = String.length s; elem_typ = Llvm_typ.i8 }
+    | Cskip size -> Llvm_typ.Array { size; elem_typ = Llvm_typ.i8 }
+    | Csingle _ -> Llvm_typ.float
+    | Cdouble _ -> Llvm_typ.double
+    | Cvec128 _ | Cvec256 _ | Cvec512 _ ->
+      Misc.fatal_error "Llvmize: vector data items not implemented"
 
   let pp_const_data_item ppf (d : Cmm.data_item) =
     match d with
-    | Cdefine_symbol _ -> assert false (* cannot happen *)
-    | Cint n -> fprintf ppf "%s" (Nativeint.to_string n)
+    | Cdefine_symbol _ | Calign _ | Csymbol_offset _ ->
+      Misc.fatal_error "Llvmize.typ_of_data_item: unexpected data item"
+    | Cint n | Cint32 n -> fprintf ppf "%s" (Nativeint.to_string n)
+    | Cint8 n | Cint16 n -> fprintf ppf "%d" n
     | Csymbol_address { sym_name; sym_global = _ } -> pp_global ppf sym_name
-    | Cint8 _ | Cint16 _ | Cint32 _ | Csingle _ | Cdouble _ | Cvec128 _
-    | Cvec256 _ | Cvec512 _ | Csymbol_offset _ | Cstring _ | Cskip _ | Calign _
-      ->
-      Misc.fatal_error "Llvmize.pp_const_data_item: not implemented"
+    | Cstring s -> fprintf ppf "c\"%s\"" s
```

Comment:
I think some kind of escaping might be
needed here (for instance if there is a
`\n` character in the string). Maybe it
will be enough to change the format
specifier to `%S`?